lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -19,9 +21,14 @@ def lookup(path, vault_url = nil)
     raise Puppet::Error, "Unable to parse a hostname from #{vault_url}" unless uri.hostname
 
     use_ssl = uri.scheme == 'https'
-    connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+    context = if use_ssl
+                Puppet::SSL::SSLContext.new
+              else
+                nil
+              end
+    connection = Puppet::Network::HttpPool.connection(uri.host, uri.port, use_ssl: use_ssl, ssl_context: context)
```

Comment:
This `connection` method is new to puppet 6.4.0; I think we should maintain compatibility with older 6.x agents for the time being, and continue using the `http_instance`.

Reply:
I have only seen, that it is deprecated, but without any versions. So when it was added only in the last release then I think it will ok, to use the old one.

Reply:
Hey @tuxmaster5000 I think this not actually resolved; I'd prefer to keep the `http_instance` method because the  `connection` code is not available until puppet 6.4.0. Can you remove your changes for this section and change the title of the PR?